prime-router/metadata/fhir_mapping/hl7/datatype/XONToOrganization.yml

```diff
@@ -0,0 +1,80 @@
+resourceType: Organization
+
+id:
+  type: STRING
+  valueOf: "GeneralUtils.generateResourceId()"
+  expressionType: JEXL
+
+name:
+  type: STRING
+  valueOf: XON.1
+  expressionType: HL7Spec
+
+extension_1-nameType:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: organization-name-type
+    valueCodeableConcept:
+      valueOf: datatype/CodeableConcept
+      generateList: true
+      expressionType: resource
+      specs: XON.2
+
+identifier_1:
+  condition: $xon10 NULL
+  valueOf: datatype/Identifier_var
+  expressionType: resource
+  vars:
+    valueIn: String, XON.3
+    code: String, XON.7
+    xon10: String, XON.10
+    checkDigit: String, XON.4
+    checkDigitScheme: String, XON.5
+    assigningAuthority: XON.6
+    location: XON.8
+  constants:
+    system: "http://terminology.hl7.org/CodeSystem/v2-0203"
+
+identifier_2:
+  condition: $xon10 NOT_NULL
+  valueOf: datatype/Identifier_var
+  expressionType: resource
+  vars:
+    valueIn: String, XON.10
+    code: String, XON.7
+    xon10: String, XON.10
+    checkDigit: String, XON.4
+    checkDigitScheme: String, XON.5
+    assigningAuthority: XON.6
+    location: XON.8
+  constants:
+    system: "http://terminology.hl7.org/CodeSystem/v2-0203"
+
+extension_1-name-representation-code:
+  condition: $value NOT_NULL
+  generateList: true
+  valueOf: extension/Extension
+  expressionType: resource
+  vars:
+    value: String, XON.9
+  constants:
+    KEY_NAME_SUFFIX: String
+    urlValue: name-representation-code
+
+extension_2-preferXon10:
```

Comment:
The 3 cases to consider are:

1. XON.3 is set and XON.10 is empty
2. XON.10 is set and XON.3 is empty
3. XON.3 and XON.10 are set

This seems to work for case 1 and 3, but not case 2. In case 2 we end up putting XON.10 into `identifier.value`, and then in the output we populate XON.3 from that and leave XON.10 blank. It's tough to fix case 2 without breaking case 1 though... We might need to have extensions for both (I would recommend using `organization-id-number` and `organization-organization-identifier` for the extension names), and still using the logic from the mapping to populate `identifier.value` in addition to just always populating the extensions if the respective field is there. Then on the FHIR->HL7 mapping, we can populate XON.3/10 from their respective extensions, with a fallback to populate just XON.10 from `identifier.value` if the extensions aren't there.

Reply:
Ah good call, yes I'll update and add a test case as well